FindPackage: Error getting the architecture from the lib directory

### What is the current behavior?

If there other files (such as hidden files) located in the NDDSHOME/lib directory, FindRTIConnextDDS will get them as architectures and an error appears:

```
-- RTI Connext DDS architecture... .DS_Store
CMake Error at rti_connext_dds-5.3.1/resource/cmake/FindRTIConnextDDS.cmake:446 (message):
  .DS_Store architecture is unsupported by this module
Call Stack (most recent call first):
  CMakeLists.txt:7 (find_package)
```

### Steps to reproduce the issue

Locate a `.DS_Store` in the lib directory under the ConnextDDS installation
Run any example using the find package

### What is the expected behavior?

Find the actual architectures and ignore hidden files

### Suggested solutions

Ignore hidden files from the found architectures

--- a/resources/cmake/FindRTIConnextDDS.cmake
+++ b/resources/cmake/FindRTIConnextDDS.cmake
@@ -656,10 +656,14 @@ if(NOT CONNEXTDDS_ARCH)
             "${CONNEXTDDS_DIR}/lib/*")
 
         foreach(architecture_name ${architectures_installed})
-            # Because the lib folder contains both target libraries and
-            # Java JAR files, here we exclude the "java" in our algorithm
-            # to guess the appropriate CONNEXTDDS_ARCH variable.
             if(architecture_name STREQUAL "java")
+                # Because the lib folder contains both target libraries and
+                # Java JAR files, here we exclude the "java" in our algorithm
+                # to guess the appropriate CONNEXTDDS_ARCH variable.
+                continue()
+            elseif(NOT "${architecture_name}" MATCHES "^[A-Za-z0-9]")
+                # Sometimes, some hidden or temporary folders/files are
+                # created. We skip them
                 continue()
             elseif("${architecture_name}" MATCHES ${CMAKE_HOST_SYSTEM_NAME})
                     if(CMAKE_HOST_SYSTEM_NAME MATCHES "Darwin")

